contract/r/gnoswap/v1/router/exact_in.gno

```diff
@@ -71,16 +72,98 @@ func ExactInSwapRoute(cur realm,
 	emission.MintAndDistributeGns(cross)
 
 	params := SwapRouteParams{
-		inputToken:  inputToken,
-		outputToken: outputToken,
-		routeArr:    routeArr,
-		quoteArr:    quoteArr,
-		deadline:    deadline,
-		typ:         ExactIn,
-		exactAmount: amountIn,
-		limitAmount: amountOutMin,
+		inputToken:        inputToken,
+		outputToken:       outputToken,
+		routeArr:          routeArr,
+		quoteArr:          quoteArr,
+		deadline:          deadline,
+		typ:               ExactIn,
+		exactAmount:       amountIn,
+		limitAmount:       amountOutMin,
+		sqrtPriceLimitX96: zeroString, // multi-hop swap is not allowed to set sqrtPriceLimitX96
 	}
 
+	return exactInSwapRoute(params, referrer)
+}
+
+// ExactInSingleSwapRoute swaps an exact amount of input tokens for output tokens through a single route.
+//
+// Executes single-hop swaps through a single specified route.
+// Allows price limit control via sqrtPriceLimitX96 parameter.
+// Applies slippage protection via minimum output amount.
+//
+// Parameters:
+//   - inputToken, outputToken: Token contract paths
+//   - amountIn: Exact input amount to swap
+//   - routeArr: Single swap route (just 1 hop)
+//   - amountOutMin: Minimum acceptable output (slippage protection)
+//   - sqrtPriceLimitX96: Price limit for swap execution (0 for no limit)
+//   - deadline: Unix timestamp for expiration
+//   - referrer: Optional referral address
+//
+// Route format:
+//   - Single-hop: "INPUT_TOKEN:OUTPUT_TOKEN:FEE"
+//
+// Returns:
+//   - amountIn: Actual input consumed
+//   - amountOut: Actual output received
+//
+// Reverts if output < amountOutMin, deadline passed, or price limit exceeded.
```

Comment:
To align with Go doc style, it would be better to change the comments in this section to full sentences.

```suggestion
// Reverts the transaction if the output amount is less than amountOutMin, 
// the deadline has passed, or the price limit is exceeded.
```